agent/cache/cache.go

```diff
@@ -237,15 +238,20 @@ type getOptions struct {
 	Fetch     func(opts FetchOptions) (FetchResult, error)
 	Info      RequestInfo
 	TypeEntry typeEntry
+	Ctx       context.Context
```

Comment:
https://golang.org/pkg/context/

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

Can we pass this in as the only other arg to `getWithIndex`? Storing it is necessary sometimes, but can lead to some pretty bad bugs.

It looks like the other callers to `getWithIndex` already have a `context.Context`, and removing this will fix the awkward calls from notify where the context is passed twice.